alfresco-transform-core-aio/alfresco-transform-core-aio/src/test/java/org/alfresco/transformer/transformers/AIOTransformRegistryTest.java

```diff
@@ -0,0 +1,301 @@
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ * -
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * -
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ * -
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.transformer.transformers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.alfresco.transform.client.model.config.TransformConfig;
+import org.alfresco.transformer.AIOTransformRegistry;
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.text.PDFTextStripper;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.core.io.ClassPathResource;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.OutputStreamWriter;
+import java.io.StringWriter;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.alfresco.transformer.transformers.TextToPdfContentTransformer.PAGE_LIMIT;
+import static org.alfresco.transformer.transformers.Transformer.TRANSFORM_NAME_PARAMETER;
+import static org.junit.Assert.*;
+
+public class AIOTransformRegistryTest
+{
+    private static final String SOURCE_MIMETYPE = "text/html";
+    private static final String TARGET_MIMETYPE = "text/plain";
+    String SOURCE_ENCODING = "sourceEncoding";
+    String TARGET_ENCODING = "targetEncoding";
+
+    AIOTransformRegistry aioTransformer = new AIOTransformRegistry();
+    ObjectMapper objectMapper = new ObjectMapper();
+
+
+    @Before
+    public void before() throws Exception
+    {
+        aioTransformer.registerTransformer(new MiscAdapter());
+        aioTransformer.registerTransformer(new TikaAdapter());
+
+    }
+
+    private void writeToFile(File file, String content, String encoding) throws Exception
+    {
+        try (OutputStreamWriter ow = new OutputStreamWriter(new FileOutputStream(file), encoding))
+        {
+            ow.append(content);
+        }
+    }
+
+    private String readFromFile(File file, final String encoding) throws Exception
+    {
+        return new String(Files.readAllBytes(file.toPath()), encoding);
+    }
+
+    private TransformConfig loadConfig(String s) throws Exception
+    {
+        return objectMapper.readValue(new ClassPathResource(s).getFile(), TransformConfig.class);
```

Comment:
Trivial:
File has to be in .../src/test/resources/ so they are duplicated across projects.  You should be able to extract them from the imported jar files using a ClassLoader. This would be similar to work added to AbstractTransformerControllerTest#getTestFile in #208

Reply:
I assume this can feed into ATS-711 ?